node_modules/gh-users/tests/test-search.js

```diff
@@ -0,0 +1,214 @@
+                jackIndex = (jackIndex === -1) ? Number.MAX_VALUE : jackIndex;
+                williamIndex = (williamIndex === -1) ? Number.MAX_VALUE : williamIndex;
+                shirleyIndex = (shirleyIndex === -1) ? Number.MAX_VALUE : shirleyIndex;
+
+                // Verify the "Jack Daniels" user is returned
+                assert.ok(jackIndex > -1);
+                assert.ok(jackIndex < Number.MAX_VALUE);
+
+                // Verify he is returned before the other two users
+                assert.ok(jackIndex < williamIndex);
+                assert.ok(jackIndex < shirleyIndex);
+                return callback();
+            });
+        });
+    });
+
+    /**
+     * Test that verifies that users can be queried on their email address
+     */
+    it('verify users can be queried on their email address', function(callback) {
+        _setup(function(jack, william, shirley) {
+
+            // Search for 'jd123' which is the start of Jack's email address
+            UsersTestsUtil.getUsers(global.tests.admins.cam2013.client, null, 'jd', 50, null, function(users) {
+                assert.ok(users.results.length > 0);
+
+                // Get the index of our three users in the results
+                var jackIndex = _.findIndex(users.results, {'id': jack.profile.id});
+                var williamIndex = _.findIndex(users.results, {'id': william.profile.id});
+                var shirleyIndex = _.findIndex(users.results, {'id': shirley.profile.id});
+
+                // Because the endpoint only returns x number of users, The "William Lawson" and "Shirley Temple"
+                // user might not get returned. Default the index value to a really high number, so we can more easily
+                // reason about them
+                jackIndex = (jackIndex === -1) ? Number.MAX_VALUE : jackIndex;
+                williamIndex = (williamIndex === -1) ? Number.MAX_VALUE : williamIndex;
+                shirleyIndex = (shirleyIndex === -1) ? Number.MAX_VALUE : shirleyIndex;
+
+                // Verify the "Jack Daniels" user is returned
+                assert.ok(jackIndex > -1);
+                assert.ok(jackIndex < Number.MAX_VALUE);
+
+                // Verify he is returned before the other two users
+                assert.ok(jackIndex < williamIndex);
+                assert.ok(jackIndex < shirleyIndex);
+                return callback();
+            });
+        });
+    });
+
+    /**
+     * Test that verifies that users can be queried on their Shibboleth id
+     */
+    it('verify users can be queried on their Shibboleth id', function(callback) {
+        // TODO
+        return callback();
+    });
+
+    /**
+     * Test that verifies that search results can be paged
+     */
+    it('verify search results can be paged', function(callback) {
+        _setup(function(jack, william, shirley) {
+
+            UsersTestsUtil.getUsers(global.tests.admins.cam2013.client, null, 'i', 2, 0, function(firstPage) {
```

Comment:
I'd rather test this on a new app (don't rely on previously created users) and verify that the paging stops after the expected number of pages and verify that the results per page are different.